### DISCUSSION TITLE
Migrate existing Expo App to solito

### DISCUSSION THREAD

**Discussion by farhan-ibrahim** [nandorojo/solito#84] [Q&A]
Hi,

I plan to migrate my existing Expo App to solito to create a web version of this app.
Let's say I have this structure in my Expo app's repo
```
...
-App.tsx
-src
---components
---navigation
---screens
-package.json
-eas.json
-app.json


```

How should I restructure the file to follow the monorepo template? Should I move everything inside `app/Expo`

**Comment by nandorojo** [comment]
you could start that way. however, ideally, you'd move everything into the `app` package and then import from there. that way it can be used by both next and expo.

i should put a migration guide in the docs

**Reply by Xizted** [reply]
Could you give an example of how to migrate an expo app to solito?

I have the following structure:

my_app/
├── assets
├── src/
│   ├── apollo
│   ├── components
│   ├── navigation
│   ├── screens
│   ├── services
│   └── utils
├── App.tsx
├── babel.config.js
├── declarations.d.ts
├── google-services.json
├── metro.config.js
├── package.json
├── tsconfig.json
└── webpack.config.js

I can't think of a neat way for the app to add the web.

**Comment by nandorojo** [comment]
I can add a guide at some point. I would 

1. create a new solito starter
2. put your source code in the `app` package
3. put your non-native dependencies in the `app` package 

for web, you then just need to re-export your screens in `apps/next/pages`

**Reply by Xizted** [reply]
Thank you very much for your response.

I already migrated the expo app to solito, it works correctly.

But, when I try to start the web, I get the following error.

```thread 'thread 'thread 'thread '<unnamed><unnamed><unnamed>' panicked at '<unnamed>' panicked at '' panicked at 'failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/node_modules/next/dist/client/router.js")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag' panicked at 'failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/node_modules/next/dist/client/next-dev.js")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diagfailed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/node_modules/next/dist/client/dev/amp-dev.js")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag', failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/node_modules/next/dist/pages/_error.js")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag', ', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs/usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs:/usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs::216:216216::216:1414:1414



note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/apps/next/pages/_app.tsx")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs:216:14
thread '<unnamed>' panicked at 'thread 'failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/apps/next/pages/_app.tsx")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag<unnamed>', ' panicked at '/usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rsfailed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/apps/next/pages/_document.tsx")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag:', 216/usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs::14216:
14
thread '<unnamed>' panicked at 'failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/node_modules/next/dist/pages/_error.js")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/swc-0.226.26/src/plugin.rs:216:14
error - ./pages/_app.tsx
Error: failed to process failed to invoke plugin: failed to invoke plugin on 'Some("/home/xizted/Programming/Tyche/Tyche/apps/next/pages/_app.tsx")'

Caused by:
    Missing export __get_transform_plugin_core_pkg_diag
```

I know it has to do with the reanimated swc plugin, but I haven't found the solution.

**Reply by nandorojo** [reply]
the SWC plug-in has to be at the exact version for Next.js. It’s very particular

**Reply by Xizted** [reply]
> the SWC plug-in has to be at the exact version for Next.js. It’s very particular

The web starts if I remove the following from the **next.config.js** file.

```
  experimental: {
    forceSwcTransforms: true,
    swcPlugins: [[require.resolve('./plugins/swc_plugin_reanimated.wasm')]],
  },
```

**Reply by nandorojo** [reply]
Yeah, I think that's due to the plugin being the wrong version. Reanimated needs a better system for this.

**Reply by Xizted** [reply]
Well... Commenting on that line works.

Two more things: 

- How to avoid that dependencies using localstorage don't give error when compiling in nextjs? 
- I need to have a navigation similar to Twitter, i.e. in the mobile version I have the navigation at the bottom and in the desktop version the navigation on one side of the screen

**Reply by nandorojo** [reply]
1. don’t import them on the server. Try next/dynamic with ssr set to false, or check if `typeof window != ‘undefined’`
2. have you seen my latest talk at next.js conf 2022?

**Reply by Xizted** [reply]
1. Done, Solved
2. No, I have not seen it.
